### ISSUE TITLE
-Wreturn-stack-address false positive

### ISSUE THREAD

**Issue by hokein** [llvm/llvm-project#120206]
https://gcc.godbolt.org/z/aMzvxerna

```
struct [[gsl::Pointer]] view {};
struct S {
    view s;
};
struct [[gsl::Owner]] Q {
    const S* get() const [[clang::lifetimebound]];
};
view g(view b, int c) {
    return c > 1 ? Q().get()->s : b;
}
```

**Comment by hokein** [comment]
If we remove the ternary operator, the diagnostic is gone, which is expected:

```
struct [[gsl::Pointer]] view {};
struct S {
    view s;
};
struct [[gsl::Owner]] Q {
    const S& get() const [[clang::lifetimebound]];
};
view g(view b, int c) {
  return Q().get().s; // no diagnostic, good.
}
```

We have a special handling to filter out this case: https://github.com/llvm/llvm-project/blob/main/clang/lib/Sema/CheckExprLifetime.cpp#L581-L584. However, the logic only works for if the `Arg` is a form of member expr (e.g. `base.ptr`).
